3266 docs rfc rewrite migrate low downtime live migration as migrate with near zero downtime (#3327)
timescale/docs: _partials/_migrate_live_run_live_migration.md

1. **Pull the live-migration docker image to you migration machine**

   ```shell

>    sudo docker pull timescale/live-migration

Reviewer: Do we need `sudo`? I could run without `sudo` on my ubuntu EC2 instance. Probably the user is already on sudoers list hence not required?

Reviewer: Always refer to the docker images with the latest release tag.
```suggestion
   sudo docker pull timescale/live-migration:v0.0.23
```

Author: Added sudo because I could not pull the image without running as Sudo on my default ec2 image. 

When I reviewed the doc, the release tag was 3 versions out. I'm sure it is best practice to specify the version, but unless we are sure to update this version number in the docs, I figured it best to go for an implicit latest. Happy to go for version number if that is best practice.

   ```
   To list the available commands, run:
   ```shell
   sudo docker run --rm -it -e PGCOPYDB_SOURCE_PGURI=$SOURCE  timescale/live-migration --help
   ```
   To see the available flags for each command, run `--help` for that command. For example:
   ```shell
   sudo docker run --rm -it -e PGCOPYDB_SOURCE_PGURI=$SOURCE  timescale/live-migration migrate --help
   ```

   
1. **Create a snapshot image of your source database in your Timescale Cloud service**

   This process checks that you have tuned your source database and target service correctly for replication, 
   then creates a snapshot of your data on the migration machine:

   ```shell
   docker run --rm -it --name live-migration-snapshot \
       -e PGCOPYDB_SOURCE_PGURI=$SOURCE \
       -e PGCOPYDB_TARGET_PGURI=$TARGET \
       --pid=host \
       -v ~/live-migration:/opt/timescale/ts_cdc \
       timescale/live-migration:v0.0.20 snapshot
   ```

>    If best practice is to make changes to the source and target, live-migration supplies information. For example:   

Reviewer: I couldn't understand this statement. What does it convey?

   ```shell
   2024-03-25T12:40:40.884 WARNING: The following tables in the Source DB have neither a primary key nor a REPLICA IDENTITY (FULL/INDEX)
   2024-03-25T12:40:40.884 WARNING: UPDATE and DELETE statements on these tables will not be replicated to the Target DB
   2024-03-25T12:40:40.884 WARNING:        - public.metrics
   Press 'c' and ENTER to continue
   ```

   If you have warnings, stop live-migration, make the suggested changes and start again.

1. **Synchronize data between your source database and your Timescale Cloud service**

    This command migrates data from the snapshot to your Timescale Cloud service, then streams 
    transactions from the source to the target. 

   ```shell
   docker run --rm -it --name live-migration-migrate \
       -e PGCOPYDB_SOURCE_PGURI=$SOURCE \
       -e PGCOPYDB_TARGET_PGURI=$TARGET \
       --pid=host \
       -v ~/live-migration:/opt/timescale/ts_cdc \
       timescale/live-migration:v0.0.20 migrate
   ```
 
   After migrating the schema, live-migration prompts you to create hypertables for tables that 
   contain time-series data in your Timescale Cloud service. Run `create_hypertable()` to convert these

>    table. For more information, see the [Hypertable docs][Hypertable docs].   

Reviewer: Create hypertable prompt is applicable only if the soutce Postgres or AWS RDS/Auora. It is not applicable for TimescaleDB or MST sources.

Reviewer: @billy-the-fish This is important.

Author: Done

   During this process, you see the migration process:

   ```shell
   Live-replay will complete in 1 minute 38.631 seconds (source_wal_rate: 106.0B/s, target_replay_rate: 589.0KiB/s, replay_lag: 56MiB)
   ```

>    If `migrate` stops add `--resume` and restart the command. 

Reviewer: ```suggestion
   If `migrate` stops add `--resume` to start from where it left.
```

Author: Done

>    Once the data is coordinated, you see the following message:

Reviewer 2: We should also mention that the user should move forward to the "hit 'c' and then ENTER" step only when the replay_lag is in few kilobytes. Otherwise, the data replication might not have finished.

Reviewer: Just to be clear, the print doesn't mean the target is in sync with source. It means, the target is *almost* caught up. Once this is seen on the console, the user has to stop the backend application writing to the source database and let the the remaining transactions to complete and get it sync with target. They can use tools like pg_top(CLI) or pg_stat_activity to view the current txn on source

Author: Done

>    Once the data is coordinated, you see the following message:

Reviewer: ```suggestion
   Once the target is almost caught-up with source, you see the following message:
```

1. **Pull the live-migration docker image to you migration machine**

   ```shell
   sudo docker pull timescale/live-migration
   ```
   To list the available commands, run:
   ```shell
   sudo docker run --rm -it -e PGCOPYDB_SOURCE_PGURI=$SOURCE  timescale/live-migration --help
   ```
   To see the available flags for each command, run `--help` for that command. For example:
   ```shell
   sudo docker run --rm -it -e PGCOPYDB_SOURCE_PGURI=$SOURCE  timescale/live-migration migrate --help
   ```

   
1. **Create a snapshot image of your source database in your Timescale Cloud service**

   This process checks that you have tuned your source database and target service correctly for replication, 
   then creates a snapshot of your data on the migration machine:

   ```shell
   docker run --rm -it --name live-migration-snapshot \
       -e PGCOPYDB_SOURCE_PGURI=$SOURCE \
       -e PGCOPYDB_TARGET_PGURI=$TARGET \
       --pid=host \
       -v ~/live-migration:/opt/timescale/ts_cdc \

>        timescale/live-migration:v0.0.20 snapshot

Reviewer: Lets use latest instead of v0.0.20

Author: Done

   ```

   Live-migration supplies information about updates you need to make to the source database and target service. For example:   

   ```shell
   2024-03-25T12:40:40.884 WARNING: The following tables in the Source DB have neither a primary key nor a REPLICA IDENTITY (FULL/INDEX)
   2024-03-25T12:40:40.884 WARNING: UPDATE and DELETE statements on these tables will not be replicated to the Target DB
   2024-03-25T12:40:40.884 WARNING:        - public.metrics
   Press 'c' and ENTER to continue
   ```

   If you have warnings, stop live-migration, make the suggested changes and start again.

1. **Synchronize data between your source database and your Timescale Cloud service**

    This command migrates data from the snapshot to your Timescale Cloud service, then streams 
    transactions from the source to the target. 

   ```shell
   docker run --rm -it --name live-migration-migrate \
       -e PGCOPYDB_SOURCE_PGURI=$SOURCE \
       -e PGCOPYDB_TARGET_PGURI=$TARGET \
       --pid=host \
       -v ~/live-migration:/opt/timescale/ts_cdc \

>        timescale/live-migration:v0.0.20 migrate

Reviewer: Lets use latest instead of v0.0.20

Author: Done
